pkg/image/api/helper.go

```diff
@@ -191,6 +178,61 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 	return ref, nil
 }
 
+// ParseDockerImageReference parses a Docker pull spec string into a
+// DockerImageReference.
+func ParseDockerImageReference(spec string) (DockerImageReference, error) {
+	// TODO replace with docker version once docker/docker PR11109 is merged upstream
+	stream, tag, id := parseRepositoryTag(spec)
+
+	ref, err := parseDockerImageReference(strings.Split(stream, "/"), spec)
+	if err != nil {
+		return ref, err
+	}
+	ref.Tag = tag
+	ref.ID = id
+
+	return ref, err
+}
+
+func ParseMultiSegmentsDockerImageReference(spec string) (DockerImageReference, error) {
```

Comment:
This should not be public - we should not have *two* DockerImageReferences in the system.  IF something is using ParseDockerImageReference that is not a valid DockerImageReference, we should create a new method which is "ParseNotQuiteDockerImageReference" (name can be better) that other code should call.  DockerImageReferences are multi segment by default, which means ParseDockerImageReference should support multi-segment.